src/backend/distributed/commands/table.c

```diff
@@ -458,6 +515,282 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand)
 }
 
 
+/*
+ * AlterTableUpdateRefTableAddDropFKeyConstraints function replaces reference table name on
+ * table(s) if we are to define foreign key constraint between a local table and
+ * a reference table. It can be the the referencing relation or the referenced
+ * relation indicated in a "constraint" field within the subcommands.
+ * If conditions are not met, this function errors out.
+ *
+ * Other than this, this function also sets skip_validation to true in foreign
+ * constraint subcommands if the referencing table not a local table
+ */
+static void
+AlterTableUpdateRefTableAddDropFKeyConstraints(AlterTableStmt *alterTableStatement)
```

Comment:
I would rename this to `PreprocessAlterTableAddDropFKey()`, since it does more than updating the relation names.

Reply:
makes more sense than its previous name, I will rename it